### PR TITLE
Schema architecture

### PR DESCRIPTION
**Pull request by brodia10**
This PR adds:
- `UserOwned` Mixin to relate objects to a User
- `Billable` Mixin for any Billable Objects like `Project`, `Task`, etc.
- Builds a solid foundation for the DB schema
- Brings the Duration and Billing components to a good working point
- Solidifies the workflow from Client to TimeEntry.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -267,6 +267,8 @@ tags
 ### Project template
 
 time_tracker/media/
+.github/*
```
Comment (mvarga6) [inline .gitignore:270]: Are you sure you want to ignore this?  This is where your github workflows would be defined, such as to run unit tests or run the linter.

**Review by mvarga6 — APPROVED** [review]
Just had one small question but looks good otherwise!

File: .gitignore (hunk 1 above)
Comment (brodia10) [reply .gitignore:270]: Good catch! We're using the tests and workflows from cookie-cutter and I think they may be a bit overkill so I wanted to ignore it for now until we have better defined workflows and PR validations.

Comment (brodia10) [reply .gitignore:270]: I'll make an issue for that now actually.

**Pull request merged**